### DISCUSSION TITLE
Using the Gemini File API for Prompts with Media

### DISCUSSION THREAD

**Discussion by giscus** [cfjedimaster/raymondcamden2023#30] [Comments]
# Using the Gemini File API for Prompts with Media

A look at how the File API can make working with multimodal prompts easier.

https://www.raymondcamden.com/2024/05/21/using-the-gemini-file-api-for-prompts-with-media

**Comment by Itzmadukz** [comment]
Hello Raymond, thanks for the demo and explanation. I was curious about how this would work in a client centred application where the image is uploaded via an input tag and extracted by handling the event via a ``` const file = event.target.files[0] ```. 

I tried building a demo of this and when i pass the 'file' from the user input as a parameter to 
```  const uploadResult = await fileManager.uploadFile(path, {
    mimeType,
    displayName: path,
  }); ```

The API returns an error '
```TypeError: (0 , import_fs.readFileSync) is not a function
    at GoogleAIFileManager.uploadFile ( ```


Would you say File manager does not work with inputs or there is a workaround. Appreciate any response. 

stack overflow question -- > https://stackoverflow.com/questions/78795207/passing-images-from-browser-input-uploads-to-googleaifilemanager-react-app

**Reply by cfjedimaster** [reply]
Sorry for the delay - I was on vacation last week. So, a couple of things come to mind.

First, note that the Gemini APIs are *probably* not something you want to use in a client side application as you would need to embed your keys. Instead, you would want to use server-side functions instead. 

I would not try to use the NPM libraries in client-side code, but instead, hit the REST APIs directly. I haven't done that for the Files API, but for the 'regular' stuff (prompt to result), it worked quite well.

**Comment by hyusetiawan** [comment]
Raymond, I have been trying to find this information to no avail, is there cost associated with uploading files to GoogleAIFileManager?

**Reply by cfjedimaster** [reply]
Based on the data here, https://ai.google.dev/gemini-api/docs/pricing, it looks to have no price. Considering it's a temporarily file storage I'm guessing they don't count it as you can only really use it with prompts, which you _do_ get charged for.
